src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -699,6 +699,25 @@ private void getMethodSig(CORINFO_METHOD_STRUCT_* ftn, CORINFO_SIG_INFO* sig, CO
         {
             MethodDesc method = HandleToObject(ftn);
 
+            // There might be a more concrete parent type specified - this can happen when inlining.
+            if (memberParent != null)
+            {
+                TypeDesc type = HandleToObject(memberParent);
+
+                // Typically, the owning type of the method is a canonical type and the member parent
+                // supplied by RyuJIT is a concrete instantiation.
+                if (type != method.OwningType)
+                {
+                    Debug.Assert(type.HasSameTypeDefinition(method.OwningType));
+                    Instantiation methodInst = method.Instantiation;
+                    method = _compilation.TypeSystemContext.GetMethodForInstantiatedType(method.GetTypicalMethodDefinition(), (InstantiatedType)type);
+                    if (methodInst.Length > 0)
+                    {
+                        method.MakeInstantiatedMethod(methodInst);
```

Comment:
The return value of `MakeInstantiatedMethod` is not used. That does not sound right.

Reply:
Sigh, good catch. I don't know if it's actually reachable for generic methods (even the non-generic method repro is very finnicky and it took me almost an hour to find something that doesn't involve LINQ and lambdas), but it's better to be safe than sorry.